apps/extension/src/state/wallets.ts

```diff
@@ -63,6 +64,35 @@ export const createWalletsSlice =
 
         return decryptedSeedPhrase.split(' ');
       },
+      reencryptSeedPhrase: async (seedPhrase: string[]) => {
+        const seedPhraseStr = seedPhrase.join(' ');
+
+        const passwordKey = get().password.key;
+        if (passwordKey === undefined) {
+          throw new Error('Password Key not in storage');
+        }
+
+        const key = await Key.fromJson(passwordKey);
+        const encryptedSeedPhrase = await key.seal(seedPhraseStr);
+
+        // Update in-memory store
+        set(state => {
+          const wallet = state.wallets.all[0];
+          if (!wallet) {
+            throw new Error('No wallet found in state');
+          }
+          wallet.custody.encryptedSeedPhrase = encryptedSeedPhrase;
```

Comment:
I don't think this works with how zustand operates, we need to instead return a new state object with this modified, right?